tests/test_api/test_course_controller.py

```diff
@@ -293,14 +294,48 @@ def test_not_authorized(self, client, fake_auth):
 
     def test_do_not_email_filter(self, client, db, admin_session):
         with test_approvals_workflow(app):
-            CoursePreference.update_opt_out(section_id=section_1_id, term_id=self.term_id, opt_out=True)
+            # Send invites them opt_out.
+            for section_id in (section_1_id, section_in_ineligible_room, section_3_id, section_4_id):
+                CoursePreference.update_opt_out(section_id=section_id, term_id=self.term_id, opt_out=True)
+
+                in_enabled_room = _is_course_in_enabled_room(section_id=section_id, term_id=self.term_id)
+                if section_id == section_in_ineligible_room:
+                    # Courses in ineligible rooms will be excluded from the feed.
+                    assert not in_enabled_room
+                else:
+                    assert in_enabled_room
+                    SentEmail.create(
+                        section_id=section_id,
+                        recipient_uids=_get_instructor_uids(section_id=section_id, term_id=self.term_id),
+                        template_type='invitation',
+                        term_id=self.term_id,
+                    )
+
+            # If course has approvals but not scheduled then it will show up in the feed.
+            Approval.create(
+                approved_by_uid=_get_instructor_uids(section_id=section_1_id, term_id=self.term_id)[0],
+                term_id=self.term_id,
+                section_id=section_1_id,
+                approver_type_='instructor',
+                publish_type_='canvas',
+                recording_type_='presentation_audio',
+                room_id=Room.get_room_id(section_id=section_1_id, term_id=self.term_id),
+            )
+            # Feed will exclude scheduled.
+            _schedule_recordings(
+                section_id=section_3_id,
+                term_id=self.term_id,
+            )
```

Comment:
I think these tests cover the reqts of https://jira.ets.berkeley.edu/jira/browse/DIABLO-175